cognite/client/data_classes/units.py

```diff
@@ -0,0 +1,178 @@
+
+    Args:
+        external_id (str): A unique identifier of the unit.
+        name (str): The name of the unit, e.g. DEG_C for Celsius.
+        long_name (str): A more descriptive name of the unit, e.g., degrees Celsius.
+        symbol (str): The symbol of the unit, e.g., °C.
+        alias_names (list[str]): List of alias names for the unit, e.g., Degree C, degC, °C, and so on.
+        quantity (str): The quantity of the unit, e.g., temperature.
+        conversion (UnitConversion): The conversion between the unit and its base unit. For example, the base unit for
+            temperature is Kelvin, and the conversion from Celsius to Kelvin is multiplier = 1, offset = 273.15.
+        source (str | None): The source of the unit, e.g., qudt.org
+        source_reference (str | None): The reference to the source of the unit, e.g., http://qudt.org/vocab/unit/DEG_C
+    """
+
+    def __init__(
+        self,
+        external_id: str,
+        name: str,
+        long_name: str,
+        symbol: str,
+        alias_names: list[str],
+        quantity: str,
+        conversion: UnitConversion,
+        source: str | None = None,
+        source_reference: str | None = None,
+    ) -> None:
+        self.external_id = external_id
+        self.name = name
+        self.long_name = long_name
+        self.symbol = symbol
+        self.alias_names = alias_names
+        self.quantity = quantity
+        self.conversion = conversion
+        self.source = source
+        self.source_reference = source_reference
+
+    def as_id(self) -> UnitID:
+        """Returns the UnitID of this unit."""
+        return UnitID(unit_external_id=self.external_id, name=self.name)
+
+    @classmethod
+    def _load(cls, resource: dict | str, cognite_client: CogniteClient | None = None) -> Self:
+        resource = json.loads(resource) if isinstance(resource, str) else resource
+        return cls(
+            external_id=resource["externalId"],
+            name=resource["name"],
+            long_name=resource["longName"],
+            symbol=resource["symbol"],
+            alias_names=resource["aliasNames"],
+            quantity=resource["quantity"],
+            conversion=UnitConversion.load(resource["conversion"]),
+            source=resource.get("source"),
+            source_reference=resource.get("sourceReference"),
+        )
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        return {
+            ("externalId" if camel_case else "external_id"): self.external_id,
+            "name": self.name,
+            ("longName" if camel_case else "long_name"): self.long_name,
+            "symbol": self.symbol,
+            ("aliasNames" if camel_case else "alias_names"): self.alias_names,
+            "quantity": self.quantity,
+            "conversion": self.conversion.dump(camel_case),
+            "source": self.source,
+            ("sourceReference" if camel_case else "source_reference"): self.source_reference,
+        }
```

Comment:
Suggestion to avoid manually dumping all instance variables:
(`from cognite.client.utils._text import convert_dict_to_case`)

```suggestion
        dumped = super().dump(camel_case)
        dumped["conversion"] = self.conversion.dump(camel_case)
        return convert_dict_to_case(dumped, camel_case)
```